sound/soc/sof/intel/hda-loader.c

```diff
@@ -602,8 +602,17 @@ int hda_dsp_post_fw_run(struct snd_sof_dev *sdev)
 		    (sdev->fw_ready.flags & SOF_IPC_INFO_D3_PERSISTENT ||
 		     sdev->pdata->ipc_type == SOF_INTEL_IPC4))
 			hdev->imrboot_supported = true;
+	} else {
+		/* reload all 3rd party module libraries during cold boot */
+		if (!hda->imrboot_supported || hda->skip_imr_boot) {
+			ret = sdev->ipc->ops->fw_loader->reload_libraries(sdev);
```

Comment:
Can this be done in higher level?
Adding a new flag hda flag to indicate that we have cold booted (or imr booted) and skip the `sof_ops(sdev)->load_library` in Intel arch code if there is no need for the reload?

Reply:
The callback should be validated, this will Oops with IPC3 after a firmware crash or when waking up from hibernation, no?